cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -315,8 +315,7 @@ class fixed_point {
    * @brief Default constructor that constructs `fixed_point` number with a
    * value and scale of zero
    */
-  CUDA_HOST_DEVICE_CALLABLE
-  fixed_point() : _value{0}, _scale{scale_type{0}} {}
+  fixed_point() = default;
```

Comment:
Note that I tried adding back the `CUDA_HOST_DEVICE_CALLABLE` but you end up with the following warnings:
```
../../../../include/cudf/fixed_point/fixed_point.hpp (318): warning: __host__ annotation is ignored on a function("fixed_point") that is explicitly defaulted on its first declaration
../../../../include/cudf/fixed_point/fixed_point.hpp (318): warning: __device__ annotation is ignored on a function("fixed_point") that is explicitly defaulted on its first declaration
```
The only reason I added the default ctor in the first place was to get some unit tests working - so I think it is fine that we leave it off.